src/System.Devices.Gpio/System.Devices.Gpio.Samples/Bme280.cs

```diff
@@ -0,0 +1,419 @@
+        // during manufacture
+        private struct Bme280_Calibration_Data
+        {
+            public ushort dig_T1;
+            public short dig_T2;
+            public short dig_T3;
+            public ushort dig_P1;
+            public short dig_P2;
+            public short dig_P3;
+            public short dig_P4;
+            public short dig_P5;
+            public short dig_P6;
+            public short dig_P7;
+            public short dig_P8;
+            public short dig_P9;
+            public byte dig_H1;
+            public short dig_H2;
+            public byte dig_H3;
+            public short dig_H4;
+            public short dig_H5;
+            public sbyte dig_H6;
+        };
+
+        private Bme280_Calibration_Data _cal_data;
+
+        private float _temperature;
+        private float _humidity;
+        private float _pressure;
+        private int _tFine;
+
+        private SpiConnectionSettings _spiSettings;
+        private SpiDevice _spiDevice;
+
+        private Pin _csPin;
+        private Pin _mosiPin;
+        private Pin _misoPin;
+        private Pin _sckPin;
+
+        /// <summary>
+        /// Use this for hardware SPI.
+        /// </summary>
+        public Bme280(Pin cs, SpiConnectionSettings spiSettings)
+        {
+            _csPin = cs;
+            _spiSettings = spiSettings;
+        }
+
+        /// <summary>
+        /// Use this for software SPI.
+        /// </summary>
+        public Bme280(Pin cs, Pin mosi, Pin miso, Pin sck)
+        {
+            _csPin = cs;
+            _mosiPin = mosi;
+            _misoPin = miso;
+            _sckPin = sck;
+        }
+
+        public void Dispose()
+        {
+            if (_spiDevice != null)
+            {
+                _spiDevice.Dispose();
+                _spiDevice = null;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the temperature calibration offset in degrees celsius.
+        /// </summary>
+        public float TemperatureCalibrationOffset { get; set; }
+
+        public void ReadSensor()
+        {
+            ReadTemperature();
+            ReadHumidity();
+            ReadPressure();
+        }
+
+        public float TemperatureInCelsius => _temperature + TemperatureCalibrationOffset;
+
+        public float TemperatureInFahrenheit => TemperatureInCelsius * 1.8F + 32;
+
+        public float Humidity => _humidity;
+
+        public float PressureInPascals => _pressure;
+
+        public float PressureInHectopascals => _pressure / 100;
+
+        public float PressureInMillibars => PressureInHectopascals;
+
+        public bool Begin()
+        {
+            Dispose();
+
+            _csPin.Mode = PinMode.Output;
+            DigitalWrite(_csPin, PinValue.High);
+
+            if (_spiSettings != null)
+            {
+                // Hardware SPI
+                _spiSettings.Mode = SpiMode.Mode0;
+                _spiSettings.ClockFrequency = 500 * 1000;
```

Comment:
"500 * 1000" could be written as 500_000;  Is the clock speed always 500KHz?

Reply:
I searched for it and it seems to support I2C up to 3.4MHz and SPI up to 10MHz. So maybe I should give the user the possibility to select the clock speed they want to use.